charts/rsyncd/templates/deployment.yaml

```
@@ -37,20 +45,26 @@ spec:
           readinessProbe:
             tcpSocket:
               port: 873
+          {{- with .Values.resources }}
           resources:
-            {{- toYaml .Values.resources | nindent 12 }}
+            {{- toYaml . | nindent 12 }}
+          {{- end }}
           volumeMounts:
-            - name: datadir
-              mountPath: {{ .Values.configuration.datadir }}
-              readOnly: true
             - name: rsyncd-conf
-              mountPath: /etc/rsyncd.conf
-              subPath: rsyncd.conf
-              readOnly: true
-            {{- if .Values.configurationFiles.jenkinsMotd.override }}
+              mountPath: /etc/rsyncd.d/rsyncd.inc
+              readOnly: true # Default for configmaps
+              subPath: rsyncd.inc
             - name: jenkins-motd
-              mountPath: {{ .Values.configuration.motd_path }}
-              subPath: jenkins-motd
+              mountPath: {{ .Values.configuration.motd.path }}
+              subPath: jenkins.motd
+              readOnly: true
+            {{- range .Values.configuration.components }}
+            - name: rsyncd-conf
```

Doesn't this name need to be unique? What happens if there is more than one component?

---

The name is internal to the Deployment. it's not a resource.

---

Not a resource OK, and sorry if I'm missing something here but WDYT of adding the `name` to it so there won't be multiple `rsyncd-conf` name entries if there are multiple components (which doesn't feel right to me)?

```suggestion
            - name: rsyncd-conf-{{ name }}
```

---

Your suggested change will break the deployment: the `name` attribute of a pod's container `volumeMount` is used to reference which volume of the same pod is projected to the mount.
https://kubernetes.io/docs/concepts/storage/volumes/ and https://kubernetes.io/docs/tasks/configure-pod-container/configure-volume-storage/ will help you about this topic.

https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#volumemount-v1-core is the reference doc about the fields, their type and goals:

> name:     This must match the Name of a Volume.


It means that if we change the name here, we also have to change it on the `volumes` section of the pod.

And the reference documentation about pod volumes states that the name must be "DNS compliant" (as usual) and must be within the pod: proof that there is no collision (as I said, it is NOT an API object but only data inside the representation of a Pod in the API).

---

Additionnaly, the name `rsync-conf` is unique but it contains multiple entries in this deployment (to avoid creating  multiple configmap entries).

I wonder if we should not to the same for the motd: adding it as en entrry of this configmap so less entries per deployment

---

Sure about the corresponding volume and volumeMount names, what's troubling me here is the fact that multiple entries of the map will have the same `name`.

I would have added the `{{ name }}`, and the same L83 + inclusion of the rsyncd-conf into the range loop.

---

> multiple entries of the map will have the same name.

This is expected: one configmap to rule them all, so same name to point to the same configmap.

Your suggestion breaks the deployments, we simply can't do it.

---

More precisely: you are pointing what is the goal of the `subPath` attribute

---

See https://github.com/jenkins-infra/helm-charts/pull/665/commits/c65598a87ed172aed8c88df9b8832ae9070e8cff